kubernetes-aws-go/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"github.com/pulumi/pulumi-awsx/sdk/go/awsx/ec2"
+	"github.com/pulumi/pulumi-eks/sdk/go/eks"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		// Get some configuration values or set default values
+		cfg := config.New(ctx, "")
+		minClusterSize, err := cfg.TryInt("minClusterSize")
+		if err != nil {
+			minClusterSize = 3
+		}
+		maxClusterSize, err := cfg.TryInt("maxClusterSize")
+		if err != nil {
+			maxClusterSize = 6
+		}
+		desiredClusterSize, err := cfg.TryInt("desiredClusterSize")
+		if err != nil {
+			desiredClusterSize = 3
+		}
+		eksNodeInstanceType, err := cfg.Try("eksNodeInstanceType")
+		if err != nil {
+			eksNodeInstanceType = "t2.medium"
+		}
+		vpcNetworkCidr, err := cfg.Try("vpcNetworkCidr")
+		if err != nil {
+			vpcNetworkCidr = "10.0.0.0/16"
+		}
+
+		// Create a new VPC, subnets, and associated infrastructure
+		eksVpc, err := ec2.NewVpc(ctx, "eks-vpc", &ec2.VpcArgs{
+			EnableDnsHostnames: pulumi.Bool(true),
+			CidrBlock:          &vpcNetworkCidr,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create a new EKS cluster
+		eksCluster, err := eks.NewCluster(ctx, "eks-cluster", &eks.ClusterArgs{
+			// Put the cluster in the new VPC created earlier
+			VpcId:                        eksVpc.VpcId,
+			// Public subnets will be used for load balancers
+			PublicSubnetIds:              eksVpc.PublicSubnetIds,
+			// Private subnets will be used for cluster nodes
+			PrivateSubnetIds:             eksVpc.PrivateSubnetIds,
+			// Change configuration values above to change any of the following settings
+			InstanceType:                 pulumi.String(eksNodeInstanceType),
+			DesiredCapacity:              pulumi.Int(desiredClusterSize),
+			MinSize:                      pulumi.Int(minClusterSize),
+			MaxSize:                      pulumi.Int(maxClusterSize),
+			// Do not give the worker nodes a public IP address
+			NodeAssociatePublicIpAddress: pulumi.Bool(false),
+			// Uncomment the next two lines for a private cluster (VPN access required)
+			// EndpointPrivateAccess: 	      pulumi.Bool(true),
+			// EndpointPublicAccess:         pulumi.Bool(false),
```

Comment:
I haven't spent much time with Kubernetes, so since I don't have a sense of common it is for someone to think about this (i.e., whether to make the cluster private or not) -- but if it's a common thing, I wonder if exposing it as an optional config setting, false by default like you have here, and then using that setting to set these values accordingly in all cases would be better than comments. That way, we'd be able to validate these properties as well, also and also be able to test both code paths if we wanted. 

Alternatively, we could leave these lines out, and handle the use case as a next-steps item in the template pages. That might give it a bit more room for explanation as well. Just some thoughts, wonder what you think.

Reply:
The "best practices" configuration would make the cluster private, but since _some_ sort of connectivity into the VPC (VPN or similar) would be required I opted against a private cluster and chose again to expose the option with a comment. I did (sort of) validate these properties by running a `pulumi preview` with them uncommented, but your suggestion to change the values and expose them uncommented may be worthwhile.